lib/gruf/prometheus/client/interceptor.rb

```diff
@@ -43,6 +43,7 @@ def call(request_context:, &block)
         #
         def send_metrics(request_context:, result:)
           prometheus_collector.started_total(request_context: request_context)
+          prometheus_collector.failed_total(request_context: request_context, result: result)
```

Comment:
This is an O(N) check here based on the class name; we probably should _first_ check for `!result.successful?` ([here](https://github.com/bigcommerce/gruf/blob/2e1eb0e29d1f9ce5551f8bdd284a8d9d081def85/lib/gruf/interceptors/timer.rb#L88)) as that will execute in O(1) time for successful requests (which we care about performance more on), before we call the failed_total call here.

(Both here and in the other interceptor)